docs-site/content/0.19.0/guide/mongodb-full-text-search.md

```diff
@@ -0,0 +1,281 @@
+# MongoDB Integration with Typesense
+
+This guide will walk you through how to integrate Typesense server with MongoDB by setting up a trigger using MongoDB's Change Streams.
+
+![Typesense MongoDB Integration Chart](@images/typesense-mongodb/mongodb.svg)
+
+## Step 1: Install and Run Typesense
+
+To install and start Typesense using docker run the following Docker command:
+
+```bash
+docker run -p 8108:8108 -v/tmp/typesense-data:/data typesense/typesense: \
+--data-dir /data --api-key=$TYPESENSE_API_KEY
+```
+
+Now, we can check if our Typesense server is ready to accept requests.
+
+```bash
+curl http://localhost:8108/health
+{"ok":true}
+```
+
+You can also run Typesense in other ways. Check out [Typesense Installation](./install-typesense.md) and [Typesense Cloud](https://cloud.typesense.org/) for more details.
+
+## Step 2: Start a MongoDB Replica Set
+
+MongoDB Replica sets provide redundancy and high availability, and are the basis for all production deployments.
+
+If you have a standalone MongoDB instance, you can convert it to a replica set by following steps:
+
+* Shutdown already running MongoDB server.
+* Start the MongoDB server by specifying -- replSet option
+```bash
+mongod --port "PORT" --dbpath "YOUR_DB_DATA_PATH" --replSet "REPLICA_SET_INSTANCE_NAME"
+```
+
+Check the status of replica set issuing the command `rs.status()` in mongo shell.
+
+## Step 3: Open a Change Stream
+
```

Comment:
Now let's open a change stream to listen for any changes to data in our MongoDB cluster. We'll later push these changes to Typesense.